### PR TITLE
DA Http Endpoints

### PR DESCRIPTION
**Pull request by bacv**
Expose `add_blob` and `get_range` from DaVerifier and DaIndexer in Nomos Http Api.

### CODE REVIEW

File: nodes/nomos-node/src/api.rs [hunk 1]
```diff
@@ -124,6 +182,11 @@ where
                 "/cryptarchia/headers",
                 routing::get(cryptarchia_headers::<T, S, SIZE>),
             )
+            .route("/da/add_blob", routing::post(add_blob::<A, B, VB, S>))
+            .route(
+                "/da/get_range",
+                routing::post(get_range::<T, C, V, VP, S, SIZE>),
+            )
```
Comment (youngjoon-lee) [inline nodes/nomos-node/src/api.rs:189]:
```suggestion
            .route(
                "/da/get_range",
                routing::get(get_range::<T, C, V, VP, S, SIZE>),
            )
```
Does this need to be `get`? (and in the related code below)

Comment (bacv) [reply nodes/nomos-node/src/api.rs:189]: Yes, `GET` here would be more idiomatic. I'm not sure if it needs to be :) Will update the app_id and range as query params.

Comment (youngjoon-lee) [reply nodes/nomos-node/src/api.rs:189]: Yeah. I don't think it's very important. Just asked to check if it was a typo or not. I'm fine with either.

**Review by youngjoon-lee — APPROVED** [review]
Looks good. I've felt that it's not easy to read long `where` clauses. But, I think it's good enough for now, and I'm not sure yet what a better way would be.

File: nomos-da/kzgrs-backend/src/common/mod.rs [hunk 2]
```diff
@@ -175,3 +176,56 @@ pub fn commitment_to_bytes(commitment: &Commitment) -> Vec<u8> {
...
+{
+    let bytes_vecs: Vec<Vec<u8>> = Deserialize::deserialize(deserializer)?;
+    bytes_vecs
+        .iter()
+        .map(|bytes| {
+            let mut cursor = Cursor::new(bytes);
+            T::deserialize_compressed(&mut cursor).map_err(serde::de::Error::custom)
+        })
+        .collect()
+}
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/src/common/mod.rs:231]: We need to check on this. There must be a better way of deriving this (probably deriving for third party crates. ). Also I think we already had this problem somewhere else. Could you please add an issue pointing to it?

**Review by danielSanchezQ — APPROVED** [review]
Looking good. Next step is to integrate it within some tests?

File: nomos-da/kzgrs-backend/src/common/mod.rs (hunk 2 above)
Comment (bacv) [reply nomos-da/kzgrs-backend/src/common/mod.rs:231]: I believe you're referring to this https://github.com/logos-co/nomos-node/pull/650#discussion_r1610021174

In this PR I've reused the same serde for Blob and Cert by moving related functions to common module. `ark_serialize` module provides `CanonicalDeserialize` trait implementations for proofs and commitments that then needs a wrapper to work with serde. Maybe you have an example of how you'd like this to be implemented?

Comment (danielSanchezQ) [reply nomos-da/kzgrs-backend/src/common/mod.rs:231]: > Maybe you have an example of how you'd like this to be implemented?

I'll give it a try after this gets merged.

**Comment by bacv** [comment]
> Looking good. Next step is to integrate it within some tests?

Yes, verifier and indexer services can now be used in out integration tests.

**Pull request merged**